LiskHQ/lisk-sdk
# Use getNodeStatus to retrieve lastBlock in HTTP API

### What was the problem?
Responses are based on the last block of the chain - so data returned by some API endpoints are not correct according to the current state while in **snapshot** mode
### How did I fix it?
By calling `modules.lastBlock.get` via a `Chain` action `getNodeStatus` instead of using the `Storage` component.
### How to test it?
Start the node in snapshot mode and check that the last block height is the current block height that is being snapshotted.
### Review checklist

* The PR resolves #3235 
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated
